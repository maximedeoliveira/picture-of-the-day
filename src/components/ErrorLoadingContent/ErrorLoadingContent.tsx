import { RotateCwIcon } from 'lucide-react-native';
import React, { ReactElement, cloneElement } from 'react';
import { StyleSheet, View } from 'react-native';
import { Button, Text, useTheme } from 'react-native-paper';

import Container from '@/components/Container/Container';

type ErrorLoadingContentProps = {
  action: () => void;
  actionTitle?: string;
  actionIcon?: ReactElement;
};

const ErrorLoadingContent = ({
  action,
  actionTitle = 'Réessayer',
  actionIcon = <RotateCwIcon />,
}: ErrorLoadingContentProps) => {
  const theme = useTheme();

  return (
    <Container justifyContent="center" alignItems="center">
      <Text>
        Une erreur s&apos;est produite lors du chargement des données.
      </Text>
      <View style={styles.buttonContainer}>
        <Button
          onPress={action}
          mode="contained"
          icon={() => cloneElement(actionIcon, { size: 18, color: 'white' })}
          dark={theme.dark}
          testID="error-button"
        >
          {actionTitle}
        </Button>
      </View>
    </Container>
  );
};

const styles = StyleSheet.create({
  buttonContainer: {
    marginTop: 12,
  },
});

export default ErrorLoadingContent;
