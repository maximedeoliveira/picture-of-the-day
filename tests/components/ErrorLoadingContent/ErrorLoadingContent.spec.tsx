import { fireEvent } from '@testing-library/react-native';
import { HandIcon } from 'lucide-react-native';
import React from 'react';

import ErrorLoadingContent from '@/components/ErrorLoadingContent/ErrorLoadingContent';

import { render } from '../../utils';

describe('ErrorLoadingComponent', () => {
  it('render component', async () => {
    const component = render(<ErrorLoadingContent action={() => {}} />);

    expect(component).toMatchSnapshot();
  });

  it('render component with custom props', () => {
    const component = render(
      <ErrorLoadingContent
        action={() => {}}
        actionTitle="Hello"
        actionIcon={<HandIcon />}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('call action props when click on button', () => {
    const mockedAction = jest.fn();
    const { getByTestId } = render(
      <ErrorLoadingContent action={mockedAction} />
    );

    fireEvent.press(getByTestId('error-button'));
    expect(mockedAction).toBeCalled();
  });
});
