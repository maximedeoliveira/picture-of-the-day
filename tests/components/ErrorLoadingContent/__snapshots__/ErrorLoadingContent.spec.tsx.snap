// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLoadingComponent render component 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "off",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "flex": 1,
      },
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "justifyContent": "center",
        "paddingHorizontal": 12,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Une erreur s'est produite lors du chargement des données.
  </Text>
  <View
    style={
      {
        "marginTop": 12,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="error-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="error-button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="error-button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginLeft": 12,
                    "marginRight": -4,
                  },
                  {
                    "marginLeft": 16,
                    "marginRight": -16,
                  },
                  false,
                ]
              }
              testID="error-button-icon-container"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="18"
                bbWidth="18"
                fill="none"
                focusable={false}
                height={18}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke="white"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 18,
                      "width": 18,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4294967295,
                      "type": 0,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth="2"
                >
                  <RNSVGPath
                    d="M21 12a9 9 0 1 1-9-9c2.52 0 4.93 1 6.74 2.74L21 8"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGPath
                    d="M21 3v5h-5"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
              testID="error-button-text"
            >
              Réessayer
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`ErrorLoadingComponent render component with custom props 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "off",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    [
      {
        "flex": 1,
      },
      {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "justifyContent": "center",
        "paddingHorizontal": 12,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    Une erreur s'est produite lors du chargement des données.
  </Text>
  <View
    style={
      {
        "marginTop": 12,
      }
    }
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "rgba(103, 80, 164, 1)",
          "borderRadius": 20,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="error-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(103, 80, 164, 1)",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="error-button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 20,
              },
            ]
          }
          testID="error-button"
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "marginLeft": 12,
                    "marginRight": -4,
                  },
                  {
                    "marginLeft": 16,
                    "marginRight": -16,
                  },
                  false,
                ]
              }
              testID="error-button-icon-container"
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="18"
                bbWidth="18"
                fill="none"
                focusable={false}
                height={18}
                meetOrSlice={0}
                minX={0}
                minY={0}
                stroke="white"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 18,
                      "width": 18,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={18}
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  propList={
                    [
                      "fill",
                      "stroke",
                      "strokeWidth",
                      "strokeLinecap",
                      "strokeLinejoin",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4294967295,
                      "type": 0,
                    }
                  }
                  strokeLinecap={1}
                  strokeLinejoin={1}
                  strokeWidth="2"
                >
                  <RNSVGPath
                    d="M18 11V6a2 2 0 0 0-2-2v0a2 2 0 0 0-2 2v0"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGPath
                    d="M14 10V4a2 2 0 0 0-2-2v0a2 2 0 0 0-2 2v2"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGPath
                    d="M10 10.5V6a2 2 0 0 0-2-2v0a2 2 0 0 0-2 2v8"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                  <RNSVGPath
                    d="M18 8a2 2 0 1 1 4 0v6a8 8 0 0 1-8 8h-2c-2.8 0-4.5-.86-5.99-2.34l-3.6-3.6a2 2 0 0 1 2.83-2.82L7 15"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "textAlign": "left",
                  },
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  [
                    {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    [
                      {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      undefined,
                    ],
                  ],
                ]
              }
              testID="error-button-text"
            >
              Hello
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
